pcsx2/gui/AppMain.cpp

```diff
@@ -531,6 +531,42 @@ extern bool FMVstarted;
 extern bool renderswitch;
 extern bool EnableFMV;
 
+void DoFmvSwitch(bool on_off)
+{
+	ScopedCoreThreadPause paused_core(new SysExecEvent_SaveSinglePlugin(PluginId_GS));
+
+	AspectRatioType& art = g_Conf->GSWindow.AspectRatio;
+	if (g_Conf->GSWindow.IsToggleAspectRatioSwitch && !EnableFMV && !FMVstarted && art == AspectRatio_4_3 && prevAR != art)
+		prevAR = art;
+
+	if (on_off) {
+		if (g_Conf->GSWindow.IsToggleAspectRatioSwitch) {
+			if (art != AspectRatio_4_3) {
+				prevAR = art;
+				art = AspectRatio_4_3;
+			}
+			if (GSFrame* gsFrame = wxGetApp().GetGsFramePtr())
```

Comment:
The part that you do in both case must be outside of the on_off if.

And you could use the else statement.
```
if (on_off){
} else {
}
```